Index: SDL-1.2.15/configure.in
===================================================================
--- configure.in
+++ configure.in
@@ -1263,35 +1263,10 @@ AC_HELP_STRING([--enable-video-directfb]
 
         DIRECTFB_REQUIRED_VERSION=0.9.15
 
-        AC_PATH_PROG(DIRECTFBCONFIG, directfb-config, no)
-        if test x$DIRECTFBCONFIG = xno; then
-            AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-            if test x$PKG_CONFIG != xno; then
-                if $PKG_CONFIG --atleast-pkgconfig-version 0.7 && $PKG_CONFIG --atleast-version $DIRECTFB_REQUIRED_VERSION directfb; then
-                    DIRECTFB_CFLAGS=`$PKG_CONFIG --cflags directfb`
-                    DIRECTFB_LIBS=`$PKG_CONFIG --libs directfb`
-                    video_directfb=yes
-                fi
-            fi
-        else
-            set -- `echo $DIRECTFB_REQUIRED_VERSION | sed 's/\./ /g'`
-            NEED_VERSION=`expr $1 \* 10000 + $2 \* 100 + $3`
-            set -- `directfb-config --version | sed 's/\./ /g'`
-            HAVE_VERSION=`expr $1 \* 10000 + $2 \* 100 + $3`
-            if test $HAVE_VERSION -ge $NEED_VERSION; then
-                DIRECTFB_CFLAGS=`$DIRECTFBCONFIG --cflags`
-                DIRECTFB_LIBS=`$DIRECTFBCONFIG --libs`
-                video_directfb=yes
-            fi
-        fi
-        if test x$video_directfb = xyes; then
-            # SuSE 11.1 installs directfb-config without directfb-devel
-            save_CFLAGS="$CFLAGS"
-            CFLAGS="$CFLAGS $DIRECTFB_CFLAGS"
-            AC_CHECK_HEADER(directfb.h, have_directfb_hdr=yes, have_directfb_hdr=no)
-            CFLAGS="$save_CFLAGS"
-            video_directfb=$have_directfb_hdr
-        fi
+        PKG_CHECK_MODULES([DIRECTFB], [directfb >= $DIRECTFB_REQUIRED_VERSION], [
+            video_directfb=yes
+		], [])
+
         AC_MSG_CHECKING(for DirectFB $DIRECTFB_REQUIRED_VERSION support)
         AC_MSG_RESULT($video_directfb)
 